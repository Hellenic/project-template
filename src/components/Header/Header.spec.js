import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import HeaderComponent from './Header';

// We need to wrap the original Header with 'MemoryRouter'
// because Header uses Link, which relies on React-Router context
const messages = {};
const Header = () => (
  <MemoryRouter>
    <IntlProvider locale="en" messages={messages}>
      <HeaderComponent />
    </IntlProvider>
  </MemoryRouter>
);

it('renders Header correctly', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
