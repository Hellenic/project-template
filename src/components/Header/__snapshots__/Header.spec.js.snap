// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly 1`] = `
<div>
  <div
    className="sc-bdvvtL eiGIhP"
  >
    <img
      alt="logo"
      className="Header-logo"
      src="react.svg"
      style={
        Object {
          "height": "80px",
        }
      }
    />
    <h2>
      Welcome to project frontend starter
    </h2>
    <h3>
      Implemented with Razzle + After.js
    </h3>
  </div>
  <a
    aria-current={null}
    href="/en/"
    onClick={[Function]}
  >
    Home
  </a>
   -
  <a
    aria-current={null}
    href="/en/about"
    onClick={[Function]}
  >
    About
  </a>
   -
  <a
    aria-current={null}
    href="/en/terms"
    onClick={[Function]}
  >
    Terms
  </a>
</div>
`;
